src/dune_pkg/dune

```diff
@@ -0,0 +1,6 @@
+(library
+ (name dune_pkg)
+ (synopsis "[Internal] Dune's packaging support")
+ (libraries stdune fiber dune_engine opam-core opam-repository opam-format)
```

Comment:
Take a look at how we vendor other libraries. We'll need to do the same for these two.

Reply:
I've added an `update-opam.sh` and vendored a modified copy. Will need to wait for https://github.com/ocaml/opam/pull/5452 to be merged to update it to the "official" repo and I also cherry-picked the changes from https://github.com/ocaml/opam/pull/4771 to make opam build with `implicit_transitive_deps false`.

Reply:
if you have any changes to vendored libraries, our convention is to fork them into opam-dune and keep the patches there.

Reply:
Ah, good, then I'll fork the stubbed-out opam (`OpamConsole`) to over there and update the script to pull it, at which point I can also remove the need for `configure.ac`.

Reply:
Done so now.